### PR TITLE
Fix #3804 Post Merge Verification

### LINKED ISSUE #3804: Post merge doesn't do check on video/image count
Well, it's pretty obvious through this thread: https://community.mybb.com/thread-224651.html , which breaks the intention of limiting maximum videos/images per post if post merge is available.

Original thread: [Post merge doesn't do check on video/image count](https://community.mybb.com/thread-224655.html)

### PR DESCRIPTION
**Pull request by Xazin**
Due to missing check on whether a post is a merge before a post is validated, it was possible to double post and merge multiple images and videos despite the limitations imposed through settings in AdminCP (Post Settings).

Resolves #3804

### CODE REVIEW

**Comment by Xazin** [comment]
The solution is rough but gives insight into a possible solution for issue #3804 .

**Comment by Xazin** [comment]
I performed some tests but hope that someone can review the changes and see if there are any problems with the logic.

File: inc/datahandlers/post.php [hunk 1]
```diff
@@ -1002,52 +1002,60 @@ function insert_post()
...
-						"posthash" => ''
+						"visible" => $visible,
+						"merge" => true
 					);
-					$db->update_query("attachments", $attachmentassign, "posthash='{$post['posthash']}' AND pid='0'");
+					
+					$plugins->run_hooks("datahandler_post_insert_merge", $this);
+					
+					return $this->return_values;
+				} else {
```
Comment (Sama34) [inline inc/datahandlers/post.php]:
```suggestion
				}
				else
				{
```

**Review by Sama34 — CHANGES_REQUESTED** [review]
https://github.com/mybb/mybb/issues/3804#issuecomment-752590036
> * if the limits are verified on the resulting posts, like currently proposed in #4214, it may be impossible to post more items within _Post Merge Time_, which may be unexpected.

Is this already accounted for in this current code? It doesn't seem like so to me.

**Comment by Xazin** [comment]
> [#3804 (comment)](https://github.com/mybb/mybb/issues/3804#issuecomment-752590036)
> 
> > * if the limits are verified on the resulting posts, like currently proposed in #4214, it may be impossible to post more items within _Post Merge Time_, which may be unexpected.
> 
> Is this already accounted for in this current code? It doesn't seem like so to me.

Yes, this fixes the issues from #3804 and the issues @dvz mentioned on that issue regarding the original pr that was made.

**Comment by euantorano** [comment]
This looks good to me and seems to work based on my testing.

**Pull request merged**